account_statement_import_online_gocardless/models/online_bank_statement_provider.py

```diff
@@ -46,6 +46,44 @@ def _get_available_services(self):
             ("gocardless", "GoCardless"),
         ]
 
+    def _gocardless_get_headers(self, basic=False):
+        """Generic method for providing the needed request headers."""
+        self.ensure_one()
+        headers = {
+            "accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        if not basic:
+            headers["Authorization"] = f"Bearer {self._gocardless_get_token()}"
+        return headers
+
+    def _gocardless_get_request(self, endpoint, params=None):
+        content = {}
+        url = url_join(GOCARDLESS_API, endpoint)
+        response = requests.get(
+            url,
+            params=params,
+            headers=self._gocardless_get_headers(),
+            timeout=REQUESTS_TIMEOUT,
+        )
+        if response.status_code == 200:
+            content = json.loads(response.text)
+        return response, content
+
+    def _gocardless_post_request(self, endpoint, data=None):
+        content = {}
+        url = url_join(GOCARDLESS_API, endpoint)
+        response = requests.post(
+            url,
+            data=data,
+            headers=self._gocardless_get_headers(basic=True),
+            timeout=REQUESTS_TIMEOUT,
+        )
+
+        if response.status_code in [200, 201]:
+            content = json.loads(response.text)
+        return response, content
```

Comment:
Simplify it in one method:

```suggestion
    def _gocardless_request(self, request_type="get", endpoint, params=None):
        content = {}
        url = url_join(GOCARDLESS_API, endpoint)
        response = getattr(request, request_type)(
            url,
            params=params,
            headers=self._gocardless_get_headers(),
            timeout=REQUESTS_TIMEOUT,
        )
        if response.status_code == 200:
            content = json.loads(response.text)
        return response, content
```

And add the extra argument of each of the other calls